code/game/gamemodes/revolution/revsquad.dm

```diff
@@ -0,0 +1,309 @@
+	if (you_are)
+		to_chat(rev_mind.current, "<span class='notice'>You are a member of the organized revolutionary organization that has infiltrated this station!</span>")
+	for(var/datum/objective/objective in rev_mind.objectives)
+		to_chat(rev_mind.current, "<B>Objective #[obj_count]</B>: [objective.explanation_text]")
+		rev_mind.special_role = "Revolutionary Squad Member"
+		obj_count++
+
+	to_chat(rev_mind.current, "<br/><b>Your fellow revolutionaries are:</b>")
+	rev_mind.store_memory("<br/><b>Your fellow revolutionaries are:</b>")
+	for(var/datum/mind/M in head_revolutionaries)
+		rev_mind.store_memory("[M.assigned_role] the [M.assigned_job.title]")
+		to_chat(rev_mind.current, "[M.assigned_role] the [M.assigned_job.title]")
+
+/datum/game_mode/proc/get_revsquad_item(var/mob/living/carbon/human/M)
+	var/possible_items = list(/obj/item/weapon/card/emag,
+														/obj/item/clothing/gloves/yellow,
+														/obj/item/weapon/gun/projectile/automatic,
+														/obj/item/device/flash/revsquad,
+														/obj/item/weapon/gun/projectile/shotgun/doublebarrel/sawnoff
+														)
+
+	var/obj/item/requisitioned = pick(possible_items)
+	if(istype(requisitioned, /obj/item/device/flash/revsquad))
+		var/obj/item/device/flash/revsquad/FR = new(M, uses = REVSQUAD_FLASH_USES)
+		requisitioned = FR
+	else
+		requisitioned = new requisitioned(M)
+	return requisitioned
+
+/datum/game_mode/proc/equip_revsquad(mob/living/carbon/human/mob)
+	if(!istype(mob))
+		return
+
+	if (mob.mind)
+		if (mob.mind.assigned_role == "Clown")
+			to_chat(mob, "Your training has allowed you to overcome your clownish nature, allowing you to wield weapons without harming yourself.")
+			mob.mutations.Remove(M_CLUMSY)
+
+	var/obj/item/T = get_revsquad_item(mob)
+
+	var/list/slots = list (
+		"backpack" = slot_in_backpack,
+		"left pocket" = slot_l_store,
+		"right pocket" = slot_r_store,
+	)
+	var/where = mob.equip_in_one_of_slots(T, slots, put_in_hand_if_fail = 0)
+
+	if (!where)
+		to_chat(mob, "The Syndicate were unfortunately unable to get you any special equipment.")
+	else
+		to_chat(mob, "The [T] in your [where] will help you to persuade the crew to join your cause.")
+		if(istype(T, /obj/item/device/flash/revsquad))
+			var/obj/item/device/flash/revsquad/FR = T
+			to_chat(mob, "<span class = 'warning'>Your [FR] has [FR.limited_conversions] uses for conversions, and not all of your comrades have one like it. Use it wisely.</span>")
+		mob.update_icons()
+		return 1
+
+/datum/game_mode/revsquad/proc/check_rev_victory()
+	for(var/datum/mind/rev_mind in head_revolutionaries)
+		for(var/datum/objective/objective in rev_mind.objectives)
+			if(!(objective.check_completion()))
+				return 0
+
+		return 1
+
+/datum/game_mode/revsquad/proc/check_heads_victory()
+	for(var/datum/mind/rev_mind in head_revolutionaries)
+		var/turf/T = get_turf(rev_mind.current)
+		if((rev_mind) && (rev_mind.current) && (rev_mind.current.stat != 2) && T && (T.z == 1))
```

Comment:
Not only is the statcheck wrongly using a bare number instead of the DEAD define, but you should use isDead() on mobs to check death.

The z-level check should also be using map.zMainStation instead of 1.